src/parser/mod.rs

```diff
@@ -11375,7 +11466,11 @@ impl<'a> Parser<'a> {
         } else {
             let object_type =
                 self.parse_one_of_keywords(&[Keyword::SEQUENCE, Keyword::SCHEMA, Keyword::TABLE]);
-            let objects = self.parse_comma_separated(|p| p.parse_object_name(false));
+            let objects = if dialect_of!(self is MySqlDialect | GenericDialect) {
+                self.parse_comma_separated(|p| p.parse_object_name_with_wildcards(false, true))
+            } else {
+                self.parse_comma_separated(|p| p.parse_object_name(false))
+            };
```

Comment:
Thinking similarly here, if there aren't any conflicting behavior we could skip the dialect check and always parse the object name with wildcard?

Reply:
There's no "conflicting behavior" per se, so the tests pass without these 2 checks, and I definitely see the value of reducing switches on dialect (especially using `dialect_of!` rather than a method on the dialect). But to be clear, this will mean that using `PostgreSqlDialect` for example will accept syntax that Postgres itself doesn't accept. If that's worth the tradeoff to reduce switching in here, I'm okay with that but just want to be clear that's okay.

Reply:
Ah yeah that would be fine, the dialect impls can accept a superset of the actual sql spec